src/Syllable.php

```diff
@@ -754,13 +754,28 @@ private function parseWord($word)
             return [$word];
         }
 
+        $wordLowerCased = mb_strtolower($word);
+
         // Is it a pre-hyphenated word?
-        if (isset($this->hyphenation[$word])) {
-            return mb_split('-', $this->hyphenation[$word]);
+        if (isset($this->hyphenation[$wordLowerCased])) {
+            $parts = [];
+            $part = '';
+            for ($i = 0, $j = 0; $i < mb_strlen($this->hyphenation[$wordLowerCased]); $i++) {
```

Comment:
Minor performance tweaks;
Return of mb_strlen(...) is constant during loop.
Prefer ++precrement over postcrement++ where possible.
To the best of my knowledge, PHP is still pretty bad at optimizing these kind of things.

Reply:
> Return of mb_strlen(...) is constant during loop.

Done.

> Prefer ++precrement over postcrement++ where possible.

Is aligned to similar code places.